### PR TITLE
Release 1.4.0

### PR DESCRIPTION
**Pull request by bastimeyer**
Let's prepare the 1.4.0 release...

Since there are so many changes, I had to add a list of notable changes instead of writing the usual text. It's better anyway, but "breaks" the "old format". The order is arbitrarily chosen and I've probably missed a few changes which should go on that list (only picked those with lots of comments and thumbs up reactions, if that makes sense).

### CODE REVIEW

**Comment by bastimeyer** [comment]
#2922 needs to be merged first

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,171 @@
...
+- Fixed: Youtube plugin ([#2858](https://github.com/streamlink/streamlink/pull/2858))
+- Fixed: Crunchyroll plugin ([#2788](https://github.com/streamlink/streamlink/pull/2788))
+- Fixed: Pixiv plugin ([#2840](https://github.com/streamlink/streamlink/pull/2840))
+- Fixed: TVplayer plugin ([#2802](https://github.com/streamlink/streamlink/pull/2802))
+- Fixed: Zattoo plugin ([#2887](https://github.com/streamlink/streamlink/pull/2887))
+- Changed: set Firefox User-Agent HTTP header by default ([#2795](https://github.com/streamlink/streamlink/pull/2795))
+- Changed: upgraded bundled FFmpeg to `4.2.2` in Windows installer ([#2916](https://github.com/streamlink/streamlink/pull/2916))
+
+
+```
```
Comment (back-to) [inline CHANGELOG.md:25]: not sure if it changes anything, but all the other ones are 
```suggestion
```text
```

**Comment by bastimeyer** [comment]
Are we good here?

**Comment by bastimeyer** [comment]
I was just writing a comment here regarding the docs, which didn't update correctly on the stable directory, even though everything was built correctly. Github must be caching the GH-pages builds when building multiple commits in a fast succession. It's working now, so ignore this comment. Just wanted to mention it, in case anybody is (or will be) wondering (on future releases).

So yeah, everything worked fine with the new deploy config :tada:

**Pull request merged**